homeassistant/components/balboa/config_flow.py

```diff
@@ -0,0 +1,96 @@
+"""Config flow for Balboa Spa Client integration."""
+from pybalboa import BalboaSpaWifi
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import callback
+
+from .const import _LOGGER, CONF_SYNC_TIME, DOMAIN
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_HOST): str, vol.Required(CONF_NAME, default="Spa"): str}
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.data[CONF_HOST] == data[CONF_HOST]:
+            raise AlreadyConfigured
+
+    _LOGGER.debug("Attempting to connect to %s", data[CONF_HOST])
+    spa = BalboaSpaWifi(data[CONF_HOST])
+    connected = await spa.connect()
+    _LOGGER.debug("Got connected = %d", connected)
+    if not connected:
+        raise CannotConnect
+    await spa.disconnect()
+
+    return {"title": data[CONF_NAME]}
```

Comment:
We want config flows to collect only the absolute minimum amount of fields to minimize the onboarding process.

Can we avoid asking them for a name and use the modelname, as they can only rename is later ?

https://github.com/garbled1/pybalboa/blob/master/pybalboa/balboa.py#L171

Reply:
I guess you could, the model names are usually pretty greusome and I doubt anyone really recognizes their spa as "BP2000G1" as opposed to just what I put in there myself like "Spa".  Mostly here, my thinking is that the users moving to this from the HACS version would just like to keep the config and have it work.

Also, I largely suspect it would be problematic if you had two spas, and they both had the same model.

Reply:
I think the use case for users with more than one spa is considerably low.